### ISSUE TITLE
[Bug]: GPU memory leak when using bad_words feature

### ISSUE THREAD

**Issue by wsp317** [vllm-project/vllm#10630]
### Your current environment

<details>
<summary>The output of `python collect_env.py`</summary>

```text
device=V100 * 4
vllm==0.6.4.post1


```

</details>

### 🐛 Describe the bug

When I attempt to use the newly added bad_words blocking feature, I encounter a GPU memory leak problem. Specifically, the memory consumption keeps increasing on GPU:0 while that of the others remains stable until an Out of Memory (OOM) error occurs.

<img width="1757" alt="image" src="https://github.com/user-attachments/assets/9f8eb7de-3d81-4016-b931-6cb441b2d873">

### Before submitting a new issue...

- [X] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

**Comment by DarkLight1337** [comment]
@njhill this might be a problem with logits processors not being garbage collected properly?

**Comment by BugCreat0r** [comment]
same problem on vllm 0.7.0, 0.7.1 and 0.7.2
